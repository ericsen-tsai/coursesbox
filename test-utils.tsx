import React, { FC, ReactElement, ReactNode } from 'react'

import { render, RenderOptions } from '@testing-library/react'

import { Themes } from './styles/themes'
import { ThemeProvider } from '@emotion/react'

const Wrapper = ({ children }: { children: ReactNode }) => {
  return <ThemeProvider theme={Themes.light}>{children}</ThemeProvider>
}

const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: Wrapper, ...options })

export * from '@testing-library/react'

export { customRender as render }
