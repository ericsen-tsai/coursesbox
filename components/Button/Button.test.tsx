import { render, screen } from '@/test-utils'
import userEvent from '@testing-library/user-event'
import Button from './index'

describe('Button test cases', () => {
  it('Render check', () => {
    const onClick = jest.fn()
    const { asFragment } = render(<Button onClick={onClick}>Button</Button>)

    expect(asFragment()).toMatchSnapshot()
  })

  it('Render check', async () => {
    const onClick = jest.fn()
    render(<Button onClick={onClick}>Button</Button>)

    const element = screen.getByRole('button')

    await userEvent.click(element)
    expect(onClick).toHaveBeenCalled()
  })
})
