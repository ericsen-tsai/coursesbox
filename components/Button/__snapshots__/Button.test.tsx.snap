// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button test cases Render check 1`] = `
<DocumentFragment>
  <button
    class="css-10nxu4j"
    color="primary"
  >
    Button
  </button>
</DocumentFragment>
`;
